content/08-mongodb/09-mongodb-indexes.mdx

```diff
@@ -0,0 +1,260 @@
+      direction: 'forward',
+      docsExamined: 50303
+    }
+  }
+}
+```
+There are several key results to focus on in this output. First, we can see in the `winningPlan` that the `stage` for this query is a `COLLSCAN`. This means that a collection scan occurred to commplete this query with a `totalDocsExamined` of 50,303 and `executionTimeMillis` of 23 milliseconds. The query had to examine every document in the collection and took 23ms even though `nReturned` was only 273 documents. While 23ms may not sound like much, this can get much longer for a collection housing one million documents.
+
+If querying on `name` is going to be a recurrent query for the application accessing this collection, we may want to create an index on this field. To do so we write the following:
+
+```
+db.comments.createIndex( {"name":1} )
+```
+
+If we take the same query with the explain planfrom before:
+
+```
+db.comments.find( { "name" : "Ramsay Bolton" } ).explain("executionStats")
+```
+we get:
+
+```
+{
+  queryPlanner: {
+    plannerVersion: 1,
+    namespace: 'sample_mflix.comments',
+    indexFilterSet: false,
+    parsedQuery: { name: { '$eq': 'Ramsay Bolton' } },
+    winningPlan: {
+      stage: 'FETCH',
+      inputStage: {
+        stage: 'IXSCAN',
+        keyPattern: { name: 1 },
+        indexName: 'name_1',
+        isMultiKey: false,
+        multiKeyPaths: { name: [] },
+        isUnique: false,
+        isSparse: false,
+        isPartial: false,
+        indexVersion: 2,
+        direction: 'forward',
+        indexBounds: { name: [ '["Ramsay Bolton", "Ramsay Bolton"]' ] }
+      }
+    },
+    rejectedPlans: []
+  },
+  executionStats: {
+    executionSuccess: true,
+    nReturned: 273,
+    executionTimeMillis: 0,
+    totalKeysExamined: 273,
+    totalDocsExamined: 273,
+    executionStages: {
+      stage: 'FETCH',
+      nReturned: 273,
+      executionTimeMillisEstimate: 0,
+      works: 274,
+      advanced: 273,
+      needTime: 0,
+      needYield: 0,
+      saveState: 0,
+      restoreState: 0,
+      isEOF: 1,
+      docsExamined: 273,
+      alreadyHasObj: 0,
+      inputStage: {
+        stage: 'IXSCAN',
+        nReturned: 273,
+        executionTimeMillisEstimate: 0,
+        works: 274,
+        advanced: 273,
+        needTime: 0,
+        needYield: 0,
+        saveState: 0,
+        restoreState: 0,
+        isEOF: 1,
+        keyPattern: { name: 1 },
+        indexName: 'name_1',
+        isMultiKey: false,
+        multiKeyPaths: { name: [] },
+        isUnique: false,
+        isSparse: false,
+        isPartial: false,
+        indexVersion: 2,
+        direction: 'forward',
+        indexBounds: { name: [ '["Ramsay Bolton", "Ramsay Bolton"]' ] },
+        keysExamined: 273,
+        seeks: 1,
+        dupsTested: 0,
+        dupsDropped: 0
+      }
+    }
+  }
+}
+```
+Comparing the same fields as the un-indexed query we see that the `stage` from the `winningPlan` is now `IXSCAN` which indicates an index was used.
```

Comment:
I read the `stage` in `winningPlan` as being `FETCH`.  Are we supposed to look at the `stage` in `winningPlan.inputStage` instead?